core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -1261,33 +1261,26 @@ private long getFlagsForStateInsertion() {
             FlagBitSets.SKIP_LOCKING |
             FlagBitSets.SKIP_XSITE_BACKUP;
 
-      boolean hasShared = false;
+      boolean hasSharedStore = false;
       acquireReadLock();
       try {
          for (CacheWriter w : nonTxWriters) {
             if (getStoreConfig(w).shared()) {
-               hasShared = true;
+               hasSharedStore = true;
                break;
             }
          }
       } finally {
          releaseReadLock();
       }
 
-      if (hasShared) {
-         if (indexShareable())
-            flags = EnumUtil.mergeBitSets(flags, FlagBitSets.SKIP_INDEXING);
-      } else {
+      if (!hasSharedStore && !configuration.indexing().isVolatile()) {
```

Comment:
The current form of this boolean condition is based on its previous + the need to re-index if the indexing provider is volatile. The two put together ... look odd to me @gustavonalle

Reply:
I've changed the ```&&``` to ```||`` and seems ok now.